accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -350,7 +350,11 @@ initializer expression and is not marked `late` is used before it is definitely
 assigned (see Definite Assignment below).
 
 It is an error if the body of a method, function, getter, or function expression
-with a potentially non-nullable return type **may complete normally**.
+with a potentially non-nullable return type **may complete normally** or may
+return without an object.
```

Comment:
I think it may be worthwhile fixing up the errors and warnings around returns, but I'm not sure this bit here is necessary, nor sufficient.

Reply:
Right, we could say that the language specification [already](https://github.com/dart-lang/language/blob/897375afe38379589ecfdffd3e7675d124545be7/specification/dartLangSpec.tex#L15434) makes it an error to have `return;` in a function unless it returns `void`, `dynamic`, or `Null`, which already implies `without an object` here; and for an `async` function [it is](https://github.com/dart-lang/language/blob/897375afe38379589ecfdffd3e7675d124545be7/specification/dartLangSpec.tex#L15474) unless `flatten(T)` is `void`, `dynamic`, or `Null`, where `T` is the declared return type.

`return;` is always OK in a generator, and that would remain so.

OK, I'll revert this change: The existing rules do actually cover all these cases already.